Cassandra-5839

mambocab opened the pull request:
Some basic tests for CASSANDRA-5839 (27596a8) along with a few improvements to `repair_test.py`:

- some style improvements (5fec1d4, 4258a8f),
- removal of [mutable default argument](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments) footguns (de5ef7e).

I do want the reviewer to sanity-check [this line](https://github.com/riptano/cassandra-dtest/blob/4258a8f832b7c0b2cc502db5196d2800fbd8e22f/repair_test.py#L310) -- I'm pretty sure we can ignore that error. It's generated [here](https://github.com/apache/cassandra/blob/d96a02a12e20bc2e0b096904ebe7a1d68532d2f3/src/java/org/apache/cassandra/db/lifecycle/LifecycleTransaction.java#L258) and appears to be a guard against invalid updates, not a fatal error.

ptnapoleon commented on repair_test.py:
```diff
@@ -216,3 +226,104 @@ def _empty_vs_gcable_no_repair(self, sequential):
+
+@since('2.2')
+class TestRepairDataSystemTable(Tester):
+    '''
```
https://www.python.org/dev/peps/pep-0257/

These should be using `"""` not `'''`

mambocab replied:
I... I never knew. Gracious!

ptnapoleon commented on repair_test.py:
```diff
@@ -216,3 +226,104 @@ def _empty_vs_gcable_no_repair(self, sequential):
+
+        Tester.setUp(self)
+        self.cluster.populate(5).start(wait_for_binary_proto=True)
+        [self.node1, _, _, _, _] = self.cluster.nodelist()
```
You can just use `self.node, = self.cluster.nodelist()` and it looks much nicer.

mambocab commented:
I addressed @ptnapoleon's comments and squashed them onto 5b7c54d. Since some of these tests flap due to SSTable GC issues, I've marked the new tests as requiring CASSANDRA-9482. With a +1 for that, I'll merge.

ptnapoleon commented:
+1
